File: Link.cpp
//
//  Link.cpp
//  LinkList
//
//  Created by Евгений Киримов on 25.08.2018.
//  Copyright © 2018 Евгений Киримов. All rights reserved.
//

#include "Link.hpp"
#include <iostream>

Link::Link(int data){
    this->data = data;
    this->next = NULL;
}

void Link::displayLink(){
    std::cout << "{" << this->data << "}";
}

Link* Link::getNext(){
    return this->next;
}

void Link::setNext(Link* link){
    this->next = link;
}

int Link::getData(){
    return this->data;
}

void Link::setData(int key){
    this->data = key;
}

File: Link.java
class Link{

    private int data;
    private Link next;


    Link(int data){
        this.data = data;
        this.next = null;
    }

    void displayLink(){
        System.out.println("{" + this.data + "}");
    }

    Link getNext(){
        return this.next;
    }

    void setNext(Link link){
        this.next = link;
    }

    int getData(){
        return this.data;
    }

    void setData(int key){
        this.data = key;
    }
}






